### PR TITLE
Upset bet

### PR DESCRIPTION
**Pull request by ReiMasa**
I believe I did the pull request correctly this time. Let me know!

### CODE REVIEW

**Comment by ReiMasa** [comment]
I would include instructions on how to get it to build on Windows, but it would require a couple more steps that may require a pc restart and some more to do. (setting path environments, putting 7zip in a particular folder)

File: extension/src/popup/popup.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ const BET_MODE_INFO = {
     elo: 'Bets using ELO of the fighters. Fighters start at 1500 ELO and use a K value of 32. Breaks ties using average bet. (<a href="https://github.com/FranciscoAT/saltyboy/blob/master/extension/src/content_scripts/bet_modes/elo.js">Source</a>)',
     eloTier:
         'Bets using the Tiered ELO of the fighters. Whenever a Fighter changes tier they go to 1500 tiered ELO and use a K value of 32. Breaks ties using average bet. (<a href="https://github.com/FranciscoAT/saltyboy/blob/master/extension/src/content_scripts/bet_modes/eloTier.js">Source</a>)',
+    upset: 'A reversed method of naive but flips the end result to the fighter who should lose instead of win.'
```
Comment (FranciscoAT) [inline extension/src/popup/popup.js:61]: I know it doesn't exist on the `master` branch just yet. But please add the "Source" link here as done in the other sections.

**Comment by FranciscoAT** [comment]
Just one comment on the MR otherwise it looks good. My other question would you be able to include some updates to the `README.md` to include a section on how to get the extension working with Windows? Would be good for the knowledge share.

**Review by FranciscoAT — CHANGES_REQUESTED** [review]

**Comment by FranciscoAT** [comment]
I'll give this a go on my machine later. Then deploy it to the store if all goes well. Code is complete barring anything I find.

**Comment by ReiMasa** [comment]
Hey, how'd it go? It still says you're requesting a revision from me but I cant tell if there is one

**Comment by FranciscoAT** [comment]
> Hey, how'd it go? It still says you're requesting a revision from me but I cant tell if there is one

Hey there! Sorry I've just been finishing up my re-work and decided to re-install my dev system haha. Been busy with those. I'll hopefully have time in a few days once I get things running and my dev system fully  stood up again!

**Comment by ReiMasa** [comment]
> > Hey, how'd it go? It still says you're requesting a revision from me but I cant tell if there is one
> 
> Hey there! Sorry I've just been finishing up my re-work and decided to re-install my dev system haha. Been busy with those. I'll hopefully have time in a few days once I get things running and my dev system fully stood up again!

No worries, thought since this is still on my git that you may have been waiting for me :)

**Comment by ReiMasa** [comment]
> > Hey, how'd it go? It still says you're requesting a revision from me but I cant tell if there is one
> 
> Hey there! Sorry I've just been finishing up my re-work and decided to re-install my dev system haha. Been busy with those. I'll hopefully have time in a few days once I get things running and my dev system fully stood up again!

I have noticed on my account that is running on my VPS that on "RNG", (simple coinflip) mode. it is abstaining from betting sometimes. Is this normal or is there a way I can debug it?

**Comment by FranciscoAT** [comment]
Hmm realistically I only tested the `naive` bet mode. Can you show me your settings. Additionally you can do some console logging for more information. You can read more about it here: https://developer.chrome.com/docs/extensions/mv3/tut_debugging/

**Comment by ReiMasa** [comment]
> Hmm realistically I only tested the `naive` bet mode. Can you show me your settings. Additionally you can do some console logging for more information. You can read more about it here: https://developer.chrome.com/docs/extensions/mv3/tut_debugging/

https://i.imgur.com/QHVJ9kr.png
https://i.imgur.com/xOb6tAb.png

Works 99% of the time from when I check, but without more info to know (since I run it 24/7) it could be less than 90%

**Comment by ReiMasa** [comment]
Also just noticed that on these settings I'm not all-inning on tournaments despite it being checked. I have no idea how this one happened. This isn't even 5% of my total.

https://i.imgur.com/6LM74OO.png

**Comment by FranciscoAT** [comment]
> > Hmm realistically I only tested the `naive` bet mode. Can you show me your settings. Additionally you can do some console logging for more information. You can read more about it here: https://developer.chrome.com/docs/extensions/mv3/tut_debugging/
> 
> https://i.imgur.com/QHVJ9kr.png https://i.imgur.com/xOb6tAb.png
> 
> Works 99% of the time from when I check, but without more info to know (since I run it 24/7) it could be less than 90%

If this is what I think it is there are one of two reasons. One ensure that no one else is running Salty Boy from within your network I believe when this happens the SaltyBet servers aren't happy about it. I remember reading something about this quite a while ago. 

The second reason is just a bit of a known unknown issue. I can't say I've tracked it down but in my time running the application there are times where it doesn't want to bet but often that happens rather infrequently and can often be resolved with a refresh.

> Also just noticed that on these settings I'm not always all-inning on tournaments despite it being checked. I have no idea how this one happened. This isn't even 5% of my total. It worked next match after with no changes
> 
> https://i.imgur.com/6LM74OO.png
> 
> But I have no problem on Naive 2% max bet set, and always all-in tournaments checked. Never fails on this one.

Sounds surprising. It could be a funky bug. Try uncheck then rechecking the "all in bet" setting. If you are running your development version can you post the `git diff`?

**Comment by ReiMasa** [comment]
> > > Hmm realistically I only tested the `naive` bet mode. Can you show me your settings. Additionally you can do some console logging for more information. You can read more about it here: https://developer.chrome.com/docs/extensions/mv3/tut_debugging/
> > 
> > 
> > https://i.imgur.com/QHVJ9kr.png https://i.imgur.com/xOb6tAb.png
> > Works 99% of the time from when I check, but without more info to know (since I run it 24/7) it could be less than 90%
> 
> If this is what I think it is there are one of two reasons. One ensure that no one else is running Salty Boy from within your network I believe when this happens the SaltyBet servers aren't happy about it. I remember reading something about this quite a while ago.
> 
> The second reason is just a bit of a known unknown issue. I can't say I've tracked it down but in my time running the application there are times where it doesn't want to bet but often that happens rather infrequently and can often be resolved with a refresh.
> 
> > Also just noticed that on these settings I'm not always all-inning on tournaments despite it being checked. I have no idea how this one happened. This isn't even 5% of my total. It worked next match after with no changes
> > https://i.imgur.com/6LM74OO.png
> > But I have no problem on Naive 2% max bet set, and always all-in tournaments checked. Never fails on this one.
> 
> Sounds surprising. It could be a funky bug. Try uncheck then rechecking the "all in bet" setting. If you are running your development version can you post the `git diff`?

Nope, I have it on a VPS from a different host, and my home computer. Correct on this though if you're placing two bets from the same IP it will discard one of them completely.

>The second reason is just a bit of a known unknown issue. I can't say I've tracked it down but in my time running the application there are times where it doesn't want to bet but often that happens rather infrequently and can often be resolved with a refresh.

Gotcha, I really only notice it happening to the one that runs on my VPS so maybe it has something to do with something with that, but I don't always monitor both of mine anyway to be sure.

> Sounds surprising. It could be a funky bug. Try uncheck then rechecking the "all in bet" setting. If you are running your development version can you post the git diff?

I'll uncheck that. It's all your current git code on my VPS though where these funky things are occurring as I've not modified that one.

**Comment by ReiMasa** [comment]
Another one just occurred that ignored settings.

It bet over the 150k limit I have set.

https://i.imgur.com/O572etl.png
https://i.imgur.com/E0cFyVh.png

**Review by FranciscoAT — APPROVED** [review]

**Pull request merged**